src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -48,4 +51,21 @@ public String getBucket() {
     return bucket;
   }
 
+  @Override
+  public void validate() {
+    super.validate();
+    if (!containsMacro("bucket") && bucket != null) {
+      String bucketName = BigQueryUtil.getBucketNameWithoutScheme(bucket);
+      if (bucketName.isEmpty()) {
+        throw new InvalidConfigPropertyException("Bucket name must be of format gs://<bucket> or <bucket>.",
```

Comment:
In prepareRun, this is actually allowed, with the bucket getting auto-created. I think we can just skip validation if it is empty.

Reply:
Wouldn't bucket be null in that case? If its not specified in the config?

Reply:
Actually I'm not sure how this can happen through the UI, probably only if the pipeline is created through REST

Reply:
This is because UI allows spaces in the bucket name. `getBucketNameWithoutScheme` method trims the spaces and can return an empty string. 
![image](https://user-images.githubusercontent.com/14131070/57107531-19f04c80-6ce5-11e9-9955-8e964656bda7.png)